src/main/java/io/axoniq/axonserver/connector/event/DcbEventChannel.java

```diff
@@ -0,0 +1,151 @@
+import io.axoniq.axonserver.grpc.event.dcb.GetTagsRequest;
+import io.axoniq.axonserver.grpc.event.dcb.GetTagsResponse;
+import io.axoniq.axonserver.grpc.event.dcb.GetTailRequest;
+import io.axoniq.axonserver.grpc.event.dcb.GetTailResponse;
+import io.axoniq.axonserver.grpc.event.dcb.SourceEventsRequest;
+import io.axoniq.axonserver.grpc.event.dcb.SourceEventsResponse;
+import io.axoniq.axonserver.grpc.event.dcb.StreamEventsRequest;
+import io.axoniq.axonserver.grpc.event.dcb.StreamEventsResponse;
+import io.axoniq.axonserver.grpc.event.dcb.TaggedEvent;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Communication channel for Event related interaction with Axon Server based on Dynamic Consistency Boundaries
+ * concept.
+ *
+ * @author Milan Savic
+ * @since 2024.1.0
+ */
+public interface DcbEventChannel {
+
+    /**
+     * Starts a new transaction to append events.
+     *
+     * @return the transaction reference onto which to register events to append along with the condition
+     */
+    AppendEventsTransaction startTransaction();
+
+    /**
+     * Provides an infinite stream of events.
+     *
+     * @param request the request used to filter events
+     * @return an infinite stream of events
+     */
+    ResultStream<StreamEventsResponse> stream(StreamEventsRequest request);
+
+    /**
+     * Provides a finite stream of events used to eventsource a model.
+     *
+     * @param request the query used to filter out events for sourcing
+     * @return the response containing events to source a model and a consistency marker to be used when trying to
+     * append new events to the event store
+     */
+    ResultStream<SourceEventsResponse> source(SourceEventsRequest request);
+
+    /**
+     * Provides tags for an event at the given global sequence.
+     *
+     * @param request the sequence of an event
+     * @return the tags fo the event
+     */
+    CompletableFuture<GetTagsResponse> tagsFor(GetTagsRequest request);
+
+    /**
+     * Provides the head of the event store.
+     *
+     * @param request empty request
+     * @return the head of the event store
+     */
+    CompletableFuture<GetHeadResponse> head(GetHeadRequest request);
+
+    /**
+     * Provides the tail of the event store.
+     *
+     * @param request empty request
+     * @return the tail of the event store
+     */
+    CompletableFuture<GetTailResponse> tail(GetTailRequest request);
+
+    /**
+     * Provides the sequence number of the event closest to the given timestamp.
+     * If the timestamp is before the first event, returns the tail sequence.
+     * If the timestamp is after the last event, returns the head sequence.
+     *
+     * @param request the request containing the timestamp to find the sequence for
+     * @return the sequence at the given timestamp
+     */
+    CompletableFuture<GetSequenceAtResponse> getSequenceAt(GetSequenceAtRequest request);
```

Comment:
Same as with the head and tail methods really: can't this just take in an `Instant` or `long` instead to construct the request for the users of this API?

Reply:
I'll change it to accept `Instant`, if we need more parameters later, we could always add an overload.